Type error in setupFastifyErrorHandler

### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Set up Fastify and TypeScript
2. Enable TypeScript feature `exactOptionalPropertyTypes`
3. Use `Sentry.setupFastifyErrorHandler`

### Expected Result

No type errors

### Actual Result

The following type error occurs:

```
error TS2345: Argument of type 'FastifyInstance<Server<typeof IncomingMessage, typeof ServerResponse>, IncomingMessage, ServerResponse<IncomingMessage>, FastifyBaseLogger, FastifyTypeProviderDefault> & PromiseLike<...> & { ...; }' is not assignable to parameter of type 'FastifyInstance'.
  The types of 'register(...).addHook' are incompatible between these types.
    Type '{ <RouteGeneric extends RouteGenericInterface = RouteGenericInterface, ContextConfig = unknown, SchemaCompiler extends FastifySchema = FastifySchema, Logger extends FastifyBaseLogger = FastifyBaseLogger, Fn extends onRequestAsyncHookHandler<...> | onRequestHookHandler<...> = onRequestHookHandler<...>>(name: "onReque...' is not assignable to type '{ (hook: string, handler: HandlerOriginal): FastifyInstance; (hook: "onError", handler: (request: FastifyRequest, reply: FastifyReply, error: Error) => void): FastifyInstance; (hook: "onRequest", handler: (request: FastifyRequest, reply: FastifyReply) => void): FastifyInstance; }'.
      Types of parameters 'hook' and 'handler' are incompatible.
        Type 'HandlerOriginal' is not assignable to type 'onRequestAsyncHookHandler<Server<typeof IncomingMessage, typeof ServerResponse>, IncomingMessage, ServerResponse<IncomingMessage>, ... 4 more ..., any> | onRequestHookHandler<...>'.
          Type '(request: FastifyRequest, reply: FastifyReply, done: HookHandlerDoneFunction) => Promise<void>' is not assignable to type 'onRequestAsyncHookHandler<Server<typeof IncomingMessage, typeof ServerResponse>, IncomingMessage, ServerResponse<IncomingMessage>, ... 4 more ..., any> | onRequestHookHandler<...>'.
            Type '(request: FastifyRequest, reply: FastifyReply, done: HookHandlerDoneFunction) => Promise<void>' is not assignable to type 'onRequestAsyncHookHandler<Server<typeof IncomingMessage, typeof ServerResponse>, IncomingMessage, ServerResponse<IncomingMessage>, ... 4 more ..., any>'.
              Target signature provides too few arguments. Expected 3 or more, but got 2.
```

See https://github.com/OliverRM/sentry-fastify-type-error for a minimal failing example.

### Product Area

Unknown

### Link

_No response_

### DSN

_No response_

### Version

_No response_

--- a/packages/node/src/integrations/tracing/fastify/index.ts
+++ b/packages/node/src/integrations/tracing/fastify/index.ts
@@ -14,7 +14,7 @@ import {
 import { generateInstrumentOnce } from '@sentry/node-core';
 import { DEBUG_BUILD } from '../../../debug-build';
 import { FastifyOtelInstrumentation } from './fastify-otel/index';
-import type { FastifyInstance, FastifyReply, FastifyRequest } from './types';
+import type { FastifyInstance, FastifyMinimal, FastifyReply, FastifyRequest } from './types';
 import { FastifyInstrumentationV3 } from './v3/instrumentation';
 
 /**
@@ -244,7 +244,7 @@ function defaultShouldHandleError(_error: Error, _request: FastifyRequest, reply
  * app.listen({ port: 3000 });
  * ```
  */
-export function setupFastifyErrorHandler(fastify: FastifyInstance, options?: Partial<FastifyHandlerOptions>): void {
+export function setupFastifyErrorHandler(fastify: FastifyMinimal, options?: Partial<FastifyHandlerOptions>): void {
   if (options?.shouldHandleError) {
     getFastifyIntegration()?.setShouldHandleError(options.shouldHandleError);
   }

--- a/packages/node/src/integrations/tracing/fastify/types.ts
+++ b/packages/node/src/integrations/tracing/fastify/types.ts
@@ -27,6 +27,15 @@ export interface FastifyInstance {
   addHook(hook: 'onRequest', handler: (request: FastifyRequest, reply: FastifyReply) => void): FastifyInstance;
 }
 
+/**
+ * Minimal type for `setupFastifyErrorHandler` parameter.
+ * Uses structural typing without overloads to avoid exactOptionalPropertyTypes issues.
+ * https://github.com/getsentry/sentry-javascript/issues/18619
+ */
+export type FastifyMinimal = {
+  register: (plugin: (instance: any, opts: any, done: () => void) => void) => unknown;
+};
+
 export interface FastifyReply {
   send: () => FastifyReply;
   statusCode: number;

